src/UserManager.ts

```diff
@@ -263,15 +263,42 @@ export class UserManager {
     }
 
     protected async _useRefreshToken(state: RefreshState): Promise<User> {
-        const response = await this._client.useRefreshToken({
-            state,
-            timeoutInSeconds: this.settings.silentRequestTimeoutInSeconds,
+        const refreshUser = async (): Promise<User> => {
+            const response = await this._client.useRefreshToken({
+                state,
+                timeoutInSeconds: this.settings.silentRequestTimeoutInSeconds,
+            });
+            return new User({ ...state, ...response });
+        };
+
+        if (!navigator.locks) {
+            // Legacy option for older browser which don't support `navigator.locks`.
+            const user = await refreshUser();
+            await this.storeUser(user);
+            this._events.load(user);
+            return user;
+        }
+
+        const broadcastChannel = new BroadcastChannel(`refresh_token_${state.refresh_token}`);
+        let user : User | null = null;
+
+        broadcastChannel.addEventListener("message", (event : MessageEvent<User>) => {
```

Comment:
The number of listener will grow every time `_useRefreshToken()` is called.

This listener should always be removed at the end, when the lock's job is done.

Reply:
Even though most recent browsers can collect observer handlers once the observed object becomes unreachable, I think it remains good practice, however, to explicitly remove them.

You can try it yourself with the example below:
- without `removeEventListener` -> 6 events received
- with `removeEventListener` -> 3 events received

```js
const _useRefreshToken = async () => {
  const broadcastChannel = new BroadcastChannel('my_key');
  let user = null;

  const onMessage = (event) => console.log('even received', event);

  broadcastChannel.addEventListener('message', onMessage);

  await navigator.locks.request(
    'my_key',
    async () => {
      if (!user) {
        await new Promise(resolve => setTimeout(resolve, 1000));
        user = {};
        broadcastChannel.postMessage('my_event');
      }
      // broadcastChannel.removeEventListener('message', onMessage);
    }
  );
};

_useRefreshToken();
_useRefreshToken();
_useRefreshToken();
```

Reply:
I would only assume it's garbage collected if the construction of a `BroadcastChannel` with at least one active listener doesn't contain hidden active handles that can't be GC'd.

In this particular case, I suspect that there is definitely a leak since the channel is never closed. At the very least, this function should close the channel with `broadcastChannel.close()` afterward.

Reply:
Okay, this is now updated.